source/agora/common/Deserializer.d

```diff
@@ -84,6 +84,10 @@ unittest
 /// Type of delegate deserializeDg
 public alias DeserializeDg = ubyte[] delegate(size_t size) @safe;
 
+/// Traits to check if a given type has an in place deserialization routine
+private enum hasDeserializeMethod (T) = is(T == struct)
```

Comment:
@bpfkorea/core in case you are confused about this weird looking declaration, this is just a shorter way of writing a full template declaration like this:

```D
template hasDeserializeMethod(T)
{
    enum hasDeserializeMethod = is(T == struct) &&
        is(typeof(T.init.deserialize(DeserializeDg.init)));
}
```

See https://dlang.org/spec/template.html#implicit_template_properties for more info.